### DISCUSSION TITLE
Detect several objects on a picture

### DISCUSSION THREAD

**Discussion by dudule73** [emgucv/emgucv#907] [General]
Hello,
I'm trying to make an app (c#) which must detect several cards shown by people to the camera (like a vote). The cards will be different QRcode printed on a paper, but framed by the same shape in order to detect them.
Card frame (object) is like this : 
![image](https://github.com/emgucv/emgucv/assets/58863019/77dadf75-bd42-48f4-a8ac-f3ba9913e00f)
Scene would be like that :
![image802](https://github.com/emgucv/emgucv/assets/58863019/79bf5666-bca7-47f3-a0b4-c4220643edca)

I'm trying to get example of this article : https://emgu.com/wiki/index.php/FAST_feature_detector_in_CSharp
but nothing is going right.

I use Emgu.cv 4.4.0.4099 (can't install more recent version under VS 2022, .net framework 4.8.1).
A lot of things have changed since the code I use as example.

My code is :
`        private void FormMain_Load(object sender, EventArgs e)
        {
            Emgu.CV.Image<Gray, Byte> img_object = null;
            Emgu.CV.Image<Gray, Byte> img_scene = null;

            try
            {
                img_object = ((Bitmap)Bitmap.FromFile("image.png")).ToImage<Gray, Byte>();
                img_scene = ((Bitmap)Bitmap.FromFile("image802.png")).ToImage<Gray, Byte>();
            }
            catch (Exception ex) { }

            Emgu.CV.Features2D.FastFeatureDetector fastFeatureDetector = new Emgu.CV.Features2D.FastFeatureDetector(50, false);
            
            MKeyPoint[] keyPoints_object, keyPoints_scene;

            keyPoints_object = fastFeatureDetector.Detect(img_object, null);
            keyPoints_scene = fastFeatureDetector.Detect(img_scene, null);

            BriefDescriptorExtractor briefDescriptorExtractor = new BriefDescriptorExtractor(32);

            Emgu.CV.Mat descriptors_object = new Emgu.CV.Mat();
            briefDescriptorExtractor.Compute(img_object, new Emgu.CV.Util.VectorOfKeyPoint(keyPoints_object), descriptors_object);

            Emgu.CV.Mat descriptors_scene = new Emgu.CV.Mat();
            briefDescriptorExtractor.Compute(img_scene, new Emgu.CV.Util.VectorOfKeyPoint(keyPoints_scene), descriptors_scene);

            Log(keyPoints_object.Length); // displaying 144
            Log(keyPoints_scene.Length); // displaying 7031

            Log(descriptors_object.GetRawData().Length); // error !!! System.NullReferenceException
            Log(descriptors_scene.GetRawData().Length);
}`
The problem is that descriptors_object returns a Mat with null data, I don't know why. But I know it can't be good...
I try to change the FastFeatureDetector parameter, but the result is the same, keyPoints_object.Length is 0 or 144.

In fact, I don't really understand what the original code does with the DescriptorExtractor...

Can someone give me explanations, or solution to achieve this ?
I need to detect the position and the rotation of all the cards in the picture.. Then I could try to read the QRCodes
Thanks for all !

**Comment by emgucv** [comment]
If you want to detect multiple QR code, you can use the WeChatQRCodeDetector:
https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Models/Dnn/WeChatQRCodeDetector.cs

based on the Emgu CV / Open CV WeChatQRCode class: https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Contrib/WechatQrcode/WeChatQRCode.cs
